lib/mongo/crypt/binding.rb

```diff
@@ -265,6 +265,8 @@ def self.setopt_log_handler(handle, log_callback)
       #   @param [ Integer ] aws_secret_access_key_len The length of the AWS
       #     secret access key (or -1 for a null-terminated string).
       #   @return [ Boolean ] Returns whether the option was set successfully.
+      #
+      #   @deprecated Use mongocrypt_setopt_kms_providers instead.
```

Comment:
This entire module if I am not mistaken is api private, therefore if these methods aren't used they should be deletable altogether.

Reply:
Removed all deprecated methods and corresponding tests. 👍